src/prefect/cli/agent.py

```diff
@@ -4,41 +4,386 @@
+)
+@click.option(
+    "--resource-manager",
+    "resource_manager_enabled",
+    is_flag=True,
+    help="Enable resource manager.",
+)
+@click.option("--rbac", is_flag=True, help="Enable default RBAC.")
+@click.option("--latest", is_flag=True, help="Use the latest Prefect image.")
+@click.option("--mem-request", help="Requested memory for Prefect init job.")
+@click.option("--mem-limit", help="Limit memory for Prefect init job.")
+@click.option("--cpu-request", help="Requested CPU for Prefect init job.")
+@click.option("--cpu-limit", help="Limit CPU for Prefect init job.")
+@click.option("--image-pull-policy", help="imagePullPolicy for Prefect init job")
+@click.option(
+    "--service-account-name", help="Name of Service Account for Prefect init job"
+)
+@click.option("--backend", "-b", help="Prefect backend to use for this agent.")
+def install(label, env, **kwargs):
+    """Generate a supervisord.conf file for a Local agent"""
+    from prefect.agent.kubernetes import KubernetesAgent
+
+    deployment = KubernetesAgent.generate_deployment_yaml(
+        labels=sorted(set(label)), env_vars=dict(e.split("=", 2) for e in env), **kwargs
+    )
+    click.echo(deployment)
+
+
+#################
+# Fargate Agent #
+#################
+
+
+@agent.group()
+def fargate():
+    """Manage Prefect Fargate agents."""
+
+
+@fargate.command(
+    context_settings=dict(ignore_unknown_options=True, allow_extra_args=True)
+)
+@add_options(COMMON_START_OPTIONS)
+@click.pass_context
+def start(ctx, **kwargs):
+    """Start a Fargate agent"""
+    from prefect.agent.fargate import FargateAgent
+
+    for item in ctx.args:
+        k, v = item.replace("--", "").split("=", 2)
+        kwargs[k] = v
+
+    start_agent(FargateAgent, **kwargs)
+
+
+#############
+# ECS Agent #
+#############
+
+
+@agent.group()
+def ecs():
+    """Manage Prefect ECS agents."""
+
+
+@ecs.command()
+@add_options(COMMON_START_OPTIONS)
+@click.option(
+    "--cluster",
+    help="The cluster to use. If not provided, your default cluster will be used",
+)
+@click.option(
+    "--launch-type",
+    type=click.Choice(["FARGATE", "EC2"], case_sensitive=False),
+    help="The launch type to use, defaults to FARGATE",
+)
+@click.option(
+    "--task-role-arn",
+    help="The default task role ARN to use for ECS tasks started by this agent.",
+)
+@click.option(
+    "--task-definition",
+    "task_definition_path",
+    help=(
+        "Path to a task definition template to use when defining new tasks "
+        "instead of the default."
+    ),
+)
+@click.option(
+    "--run-task-kwargs",
+    "run_task_kwargs_path",
+    help="Path to a yaml file containing extra kwargs to pass to `run_task`",
+)
+def start(**kwargs):
+    """Start an ECS agent"""
+    from prefect.agent.ecs import ECSAgent
+
+    start_agent(ECSAgent, **kwargs)
+
+
+#############################
+# Deprecated Agent Commands #
+#############################
+
+_agents = {
+    "fargate": "prefect.agent.fargate.FargateAgent",
```

Comment:
Since it seems `prefect agent start fargate` is still supported, would this be a good time to transition this directly to `prefect agent ecs start` instead of going through a second deprecation?

Reply:
I don't want to start raising a deprecation warning about the fargate agent until we're sure the ECS agent is 100% ready to take over all user's use cases. I think 2 deprecation cycles is fine:

- start warning users over to the new CLI pattern, which should be easy on their end
- once docs are up, start warning users over to the ECS agent, which is a much more involved move